src/lib/hdlc/hdlc.cpp

```diff
@@ -136,14 +136,14 @@ otError Encoder::Encode(uint8_t aByte)
 
     if (HdlcByteNeedsEscape(aByte))
     {
-        VerifyOrExit(mWritePointer.CanWrite(2), error = OT_ERROR_NO_BUFS);
+        ACTION_IF_NOT(mWritePointer.CanWrite(2), error = OT_ERROR_NO_BUFS);
```

Comment:
It may not be clear from `ACTION_IF_NOT` that it will exit. How about `ENSURE_ACTION()` to be consisitent with other ones?

Reply:
Or maybe we can define a single `ENSURE()` which requires an action statement and then define a `NO_ACTION` macro to pass in cases where there is no action. 

I think there not that many cases where we don't have any action? This way the more common case will uses a shorter name `ENSURE()`.